src/functional/iterator/embedded.py

```diff
@@ -478,7 +472,9 @@ def promote_scalars(val: CompositeOfScalarOrField):
     """Given a scalar, field or composite thereof promote all (contained) scalars to fields."""
     if isinstance(val, tuple):
         return tuple(promote_scalars(el) for el in val)
-    val_type = xtyping.infer_type(val)
+    val_type = cast(
+        npt.DTypeLike, xtyping.infer_type(val)
+    )  # TODO(havogt) how can we assert that something is DTypeLike?
```

Comment:
```suggestion
    val_type = xtyping.infer_type(val)
    assert issubclass(val_type, (np.generic, int, float))
```
This should be fine for all practical purposes (not great though). Hopefully the cast is also superfluous after the assert.

Reply:
done as typeguard